hipscat/util.py

```diff
@@ -289,148 +286,111 @@ def xmatchmap_dict(hp_match_map):
     return data
 
 
-def catalog_columns_selector_withdtype(cat_md, cols):
+def rename_meta_cols(c1_cols, c2_cols, suffix='_2'):
+    '''
+        finds columns with the same name and appends
+        a suffix to the column name. Should only be applied
+        to the c2_cols dictionary
     '''
-        Establish the return columns for the dataframe's metadata
-         dask.dataframe.map_partitions() requires the metadata of the resulting 
-         dataframe to be defined prior to execution. The column names and datatypes
-         are defined here and passed in the 'meta' variable
+    for c2c in c2_cols:
+        if c2c in c1_cols:
+            c2c_i = c2_cols.index(c2_cols[c2c])
+            c2_cols[c2c_i] = f'{c2c}{suffix}'
+    return c2_cols
 
-         it expects the ra_kw, dec_kw, and id_kw fields with their respective dtypes
-         if the user want's other columns, it will append them
 
-         TODO: validate user-defined cols fields {key: dtype} 
+def cater_input_cols(cols, cat_md):
```

Comment:
Cater?

This looks like a set union operation.

Reply:
changed function name to `validate_user_input_cols`